source/ScreenSaverHero.hx

```diff
@@ -0,0 +1,67 @@
+package;
+
+import flixel.util.FlxSpriteUtil;
+import flixel.FlxG;
+import flixel.FlxSprite;
+
+class ScreenSaverHero extends FlxSprite
+{
+	var speed = 50;
+
+	public function new(x:Float = 0, y:Float = 0)
+	{
+		super(x, y);
+
+		makeGraphic(20, 10);
+		controlVelocity();
+	}
+
+	override function update(elapsed:Float)
+	{
+		if (x >= 0 && x < 91)
+			color = 0xFF0000;
+		else if (x >= 91 && x < 182)
+			color = 0xFF7F00;
+		else if (x >= 182 && x < 273)
+			color = 0xFFFF00;
+		else if (x >= 273 && x < 364)
+			color = 0x00FF00;
+		else if (x >= 364 && x < 455)
+			color = 0x0000FF;
+		else if (x >= 455 && x < 546)
+			color = 0x4B0082;
+		else
+			color = 0x9400D3;
```

Comment:
There are definitely magic numbers here - it's not necessarily going to look that much cleaner if we make a bunch of variables for all of these arbitrary x values, though, right?

I can some think of some ways to make this cleaner - not required for merging the PR, but it'd be worth considering.

to give an example from me, I imagined we could do something like 
```haxe
var rainbow_colors = (create an array containing all of your 7 hard-coded numbers)
// Figure out what "section" of the screen we're generally in. This ends up being a float
var region_as_a_float = x / (screen_width / rainbow_colors.length)
// truncate the number to an integer, then make sure we didn't end up larger than our array size
var region_as_an_int = Math.min(Std.int(region_as_a_float), rainbow_colors.length - 1)
color = rainbow_colors[region_as_an_int]
```

Reply:
I did something similar here.